test/integration/logic/TypeInferenceTest.java

```diff
@@ -188,6 +191,30 @@ public void is_inference() throws IOException {
         assertEquals(expectedExhaustive, resolvedTypeMap(disjunction.conjunctions().get(0)));
     }
 
+    @Test
+    public void iid_inference() throws IOException {
+        define_standard_schema("basic-schema");
+        transaction.commit();
+        session.close();
+        session = typedb.session(database, DATA);
+        transaction = session.transaction(WRITE);
+        Entity person = transaction.concepts().getEntityType("person").create();
+
+        TypeInference typeInference = transaction.logic().typeInference();
+
+        // using a person IID, the attribute can only be a name or email, but not a dog label
+        String queryString = "match $p iid " + person.getIID().toHexString() + "; $p has $a;";
+        Disjunction disjunction = createDisjunction(queryString);
+        typeInference.infer(disjunction);
+
+        Map<String, Set<String>> expected = new HashMap<>() {{
+            put("$p", set("person"));
+            put("$a", set("name", "email"));
+        }};
+
```

Comment:
test that the IID type information restricts both the `$p` and by connection the attributes we search for